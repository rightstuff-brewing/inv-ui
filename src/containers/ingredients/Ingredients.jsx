import React from 'react';
import {Button, Table} from 'react-bootstrap';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
    createIngredient
} from '../../modules/ingredients';
import NewIngredient from './NewIngredient';

import './Ingredients.css'

class Ingredients extends React.Component {
    constructor(props) {
        super(props);
        this.state = {showModal: false};
        this.close = this.close.bind(this);
        this.open = this.open.bind(this);
        this.submitIngredient = this.submitIngredient.bind(this);
    }

    close() {
        this.setState({ showModal: false });
    }

    open() {
        this.setState({ showModal: true });
    }

    submitIngredient(input) {
        this.props.createIngredient(input);
        if (this.state.showModal) {
            this.close();
        }
    }   
    
    render() {
        return (
            <div>
                <Table striped bordered condensed hover>
                    <thead>
                        <tr>
                            <th className="center">Ingredient</th>
                            <th className="center">Type</th>
                            <th className="center">URL</th>
                            <th className="center">Admin Actions</th>
                        </tr>
                    </thead>
                    <tbody>
                        {/* Traverse ingredients array */}
                        {this.props.ingredients.map((ingredient, i) => (
                        <tr key={i}>
                            <td>{ingredient.name}</td>
                            <td>{ingredient.type.displayName}</td>
                            <td className="center">
                                <a href={ingredient.url} target="_blank">URL</a>
                            </td>
                            <td className="right">
                                <Button bsStyle="primary">Edit</Button>
                                <Button bsStyle="danger">Delete</Button>
                            </td>
                        </tr>
                        ))}
                    </tbody>
                </Table>
                <br />
                <Button bsStyle="success" onClick={this.open}>Create</Button>

                <NewIngredient show={this.state.showModal} close={() => this.close()} submit={(input) => this.submitIngredient(input)} />
            </div>
        );
    }
}

const mapStateToProps = state => ({
    ingredients: state.ingredients.ingredients
});

const mapDispatchToProps = dispatch => bindActionCreators({
    createIngredient
}, dispatch);

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Ingredients);