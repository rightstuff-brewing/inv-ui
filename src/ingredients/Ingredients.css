table, th, td {
    border: 1px solid black;
}

table {
    width: 100%;
}

.center {
    text-align: center;
}