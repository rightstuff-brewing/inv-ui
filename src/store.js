import { createStore, applyMiddleware, compose } from 'redux';
import { routerMiddleware } from 'react-router-redux';
import { reactReduxFirebase } from 'react-redux-firebase';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import rootReducer from './modules';

export const history = createHistory();

const firebaseConfig = {
  apiKey: "AIzaSyBzQefr9YyRjj0CpoL9qk7RHPu1plv_Vkg",
  authDomain: "rightstuff-176212.firebaseapp.com",
  databaseURL: "https://rightstuff-176212.firebaseio.com",
  projectId: "rightstuff-176212",
  storageBucket: "rightstuff-176212.appspot.com",
  messagingSenderId: "183311933229"
};
const rrfConfig = { userProfile: 'users' };

const initialState = {};
const enhancers = [];
const middleWare = [
  thunk,
  routerMiddleware(history)
];

if (process.env.NODE_ENV === 'development') {
  const devToolsExtension = window.devToolsExtension;

  if(typeof devToolsExtension === 'function') {
    enhancers.push(devToolsExtension);
  }
}

const composedEnhancers = compose(
  applyMiddleware(...middleWare),
  reactReduxFirebase(firebaseConfig, rrfConfig),
  ...enhancers
);

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers
);

export default store;