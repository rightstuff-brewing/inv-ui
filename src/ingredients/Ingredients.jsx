import React from 'react';
import './Ingredients.css'

function Ingredients() {
    return (
        <table>
            <thead>
                <tr>
                    <th>Ingredient</th>
                    <th>Type</th>
                    <th></th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td>Crisp Malt Wheat</td>
                    <td>Malt</td>
                    <td>
                        <button>Edit</button>
                        <button>Delete</button>
                    </td>
                </tr>
            </tbody>
        </table>
    );
}

export default Ingredients;