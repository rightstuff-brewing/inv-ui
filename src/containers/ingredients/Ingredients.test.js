import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store, { history } from '../../store';
import Ingredients from './Ingredients';

describe('Ingredients component', () => {
  it('matches the snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Ingredients />
      </Provider>      
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
