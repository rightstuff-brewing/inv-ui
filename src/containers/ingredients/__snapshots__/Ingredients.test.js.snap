// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients component matches the snapshot 1`] = `
<div>
  <table
    className="table table-striped table-bordered table-condensed table-hover"
  >
    <thead>
      <tr>
        <th
          className="center"
        >
          Ingredient
        </th>
        <th
          className="center"
        >
          Type
        </th>
        <th
          className="center"
        >
          URL
        </th>
        <th
          className="center"
        >
          Admin Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Crisp Malt Wheat
        </td>
        <td>
          Malt
        </td>
        <td
          className="center"
        >
          <a
            href="https://www.weyermann.de/eng/gelbe_seiten_en.asp?snr=1&idkat=1013&umenue=yes&idmenue=37&sprache=2"
            target="_blank"
          >
            URL
          </a>
        </td>
        <td
          className="right"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            type="button"
          >
            Edit
          </button>
          <button
            className="btn btn-danger"
            disabled={false}
            type="button"
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>
  <br />
  <button
    className="btn btn-success"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Create
  </button>
</div>
`;
